python/cudf/cudf/pylibcudf_tests/test_binaryops.py

```diff
@@ -114,90 +110,111 @@ def get_result():
     ids=idfn,
 )
 def test_add(pa_data, plc_data):
-    def add(x, y):
-        if x is None or y is None:
-            return None
-        return x + y
-
     _test_binaryop_inner(
         pa_data,
         plc_data,
-        add,
+        pa.compute.add,
         plc.binaryop.BinaryOperator.ADD,
     )
 
 
 @pytest.mark.parametrize(
     "pa_data",
-    [("int64", "int64", "int64"), ("int64", "float64", "float64")],
+    [
+        ("int64", "int64", "int64"),
+        ("int64", "float64", "float64"),
+        ("int64", "int64", "datetime64[ns]"),
+    ],
```

Comment:
Why can these parametrizations not be applied directly to the top-level `pa_data` fixture, which would avoid repeating them here?

Ah: I see that some of the tests don't apply to all the different dtype tuples. Is that to reduce the number of tests for cases where the binop makes no sense?

Reply:
Right, I went with this approach because we decided we shouldn't be checking the entire lhs-rhs-out-op 4d matrix, so the latest refactor has been aimed at putting us in a position where its as easy as possible to dictate what cases we feel are important for each operator.

Reply:
OK, nothing to do here then.